### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by phphuc612** [shengqiliu1/SewingLDM#4]
Dear authors,
Congratulation on your amazing work. 
I am trying to infer from a representative example but there seems a significant mismatch in design pattern, leading to poor performance when reproducing metrics.
This sample is drawn from GarmentCodeData, then rendered without smpl and finally passed into [PiDiNet](https://github.com/hellozhuo/pidinet?tab=readme-ov-file) to extract sketch.
I would greatly appreciate it if you could advise me on whether any steps should be adjusted in order to obtain the results reported in your paper.

Render images:
<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/0c4b6d50-cccc-45c2-af75-6bfb744003ef" />
<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4656744e-4907-47b7-8590-14d72e5b6e77" />

Sketches:
<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/02598886-624b-45ed-b188-7eb1ff8072ed" />
<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/3bee211b-5b26-456e-bd58-5c8701385ccd" />

GT pattern (from test set):
<img width="898" height="494" alt="Image" src="https://github.com/user-attachments/assets/7e6bd9a9-f371-411f-a4b8-faffcd676f4a" />

Inferred pattern:
<img width="2162" height="362" alt="Image" src="https://github.com/user-attachments/assets/1cd9ee68-c3a4-4c7f-a201-f1785642ff14" />

**Comment by shengqiliu1** [comment]
To align with the sketches commonly used in real-world scenarios, the clothing is rendered separately, using fully frontal and fully back-view sketches. The sketch we employ is shown in the figure below:

<img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/0590dfa8-1a8d-4565-8927-5e2b96e52d16" />

<img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/7f5d6a24-32d8-4301-9c4e-0b840cece7b7" />

The sewing pattern generated solely under sketch guidance is presented as follows.

<img width="984" height="485" alt="Image" src="https://github.com/user-attachments/assets/c7b86e87-b405-4b47-bd88-dea149a78694" />

**Comment by phphuc612** [comment]
Hi @shengqiliu1,
Thank you for your timely response.

I have modified the rendering according to your instructions, but the quality still does not seem to improve. For the sake of reproducibility, I would greatly appreciate it if you could share your rendering code/evaluation code.

Another follow-up question, how can I make the inference of your method more robust when there is hardly any visual difference 
Once again, I am grateful for your work and time to help me.

----------
## Sample 1:
Input:
<img width="3342" height="853" alt="Image" src="https://github.com/user-attachments/assets/f01b68c4-4718-460f-b715-2b0b0cf28cc6" />
Output:

<img width="2324" height="360" alt="Image" src="https://github.com/user-attachments/assets/71540084-1cee-4f4b-bd8e-b152c21453ff" />

Groundtruth:
<img width="898" height="494" alt="Image" src="https://github.com/user-attachments/assets/26891365-1e3f-4e45-abe8-da335081f449" />

---------
## Sample 2:
Input:
<img width="3132" height="864" alt="Image" src="https://github.com/user-attachments/assets/e66d2c8d-d812-4dde-9225-530b4e580cc7" />
Output:
<img width="702" height="380" alt="Image" src="https://github.com/user-attachments/assets/46aca008-b382-4522-9561-db4d819ab43a" />
Groundtruth:

<img width="775" height="449" alt="Image" src="https://github.com/user-attachments/assets/dd28dd1e-45d0-44a3-89fe-428ea91eb942" />

**Comment by shengqiliu1** [comment]
Indeed, the difference in the rendering method has caused discrepancies in the sketches. Currently, I am occupied with other tasks and need more time to organize the code. For out-of-domain sketch data, you can add text conditions. Due to the training setup, our model will prioritize the text conditions before adhering to the sketch conditions.

**Comment by phphuc612** [comment]
Thank you for your response. 
I will keep this issue open and update following your instructions.

**Comment by OwlNjust** [comment]
I am also in the process of reproducing this excellent work and have encountered a similar issue during the evaluation stage. Based on my attempts, I believe the root cause is that the description of the input text condition is not sufficiently detailed. However, overly detailed text conditions make the generated samples unusable, so a balance needs to be struck in practical testing.
